lua/resession/init.lua

```diff
@@ -438,18 +438,18 @@ M.load = function(name, opts)
   -- Ignore all messages (including swapfile messages) during session load
   local shortmess = vim.o.shortmess
   vim.o.shortmess = "aAF"
-  if not data.tab_scoped then
-    -- Set the options immediately
-    util.restore_global_options(data.global.options)
-    vim.cmd(string.format("cd %s", data.global.cwd))
-  end
   local scale = {
     vim.o.columns / data.global.width,
     (vim.o.lines - vim.o.cmdheight) / data.global.height,
   }
-  for _, buf in ipairs(data.buffers) do
+  for i, buf in ipairs(data.buffers) do
     local bufnr = vim.fn.bufadd(buf.name)
     if buf.loaded then
+      -- Set global options before loading the first buffer
+      if i == 1 and not data.tab_scoped then
+        util.restore_global_options(data.global.options)
+        vim.api.nvim_set_current_dir(data.global.cwd)
+      end
```

Comment:
This doesn't feel like the right way to do this. If the first buffer in the restore list was unloaded, then we will _never_ restore the global options.

Could this be fixed by just adding another
```lua
if not data.tab_scoped then
  vim.api.nvim_set_current_dir(data.global.cwd)
end
```
after the tabs and buffers are all restored?

Reply:
Yeah nvm, I found an edgecase where if one of the buffers in the restore list is not under the cwd, then the relative file path does not show correctly for all the subsequent buffers, even those buffers are under the cwd. Removing the check for the first buffer fixes this, but it feels redundant to repeatedly set global options for each buffer. 

I'll try out your suggestion

Reply:
@stevearc setting the cwd again after loading all buffers/tabs seems to work